app/vmselect/vmui.go

```diff
@@ -37,7 +37,7 @@ type panelSettings struct {
 	Unit        string   `json:"unit,omitempty"`
 	Expr        []string `json:"expr"`
 	Alias       []string `json:"alias,omitempty"`
-	ShowLegend  bool     `json:"showLegend,omitempty"`
+	ShowLegend  *bool    `json:"showLegend"`
```

Comment:
Common practice is to use default type values as defaults and name param in order to change behavior. 

 For instance `ShowLegend` is boolean with default value as false.

 If you rename it to `DisableLegend` it will have default value as false and only explicit change it will modify behavior. 

 Any way, it's too late to change it.